src/ConvertToPriceInfoBeta.cpp

```diff
@@ -0,0 +1,91 @@
+#include "ConvertToPriceInfoBeta.h"
+#include <algorithm>
+#include <deque>
+#include <iterator>
+#include <unordered_set>
+
+std::pair<PriceInfoBeta, PriceInfoBeta> ConvertToPriceInfoBeta::convert_to_PriceInfoBeta_v1(PriceInfo info, std::string crypto){
+    auto temp = info.symbol;
+
+    PriceInfoBeta beta1;
+    beta1.from=crypto;
+    beta1.to=temp.substr(crypto.size(), temp.size());
+    beta1.bid_price=info.bid_price;
+    beta1.bid_quantity=info.bid_quantity; 
+
+    PriceInfoBeta beta2;
+    beta2.from=temp.substr(crypto.size(), temp.size());
+    beta2.to=crypto; 
+    beta2.bid_price=info.ask_price;
+    beta2.bid_quantity=info.ask_quantity;
+    return {beta1, beta2};
+}
+
+
+std::pair<PriceInfoBeta, PriceInfoBeta> ConvertToPriceInfoBeta::convert_to_PriceInfoBeta_v2(PriceInfo info, std::string crypto, int pos){
+    auto temp = info.symbol;
+    
+    PriceInfoBeta beta1;
+    beta1.from=crypto;
+    beta1.to=temp.substr(0, pos);
+    beta1.bid_price=info.ask_price;
+    beta1.bid_quantity=info.ask_quantity; 
+
+    PriceInfoBeta beta2;
+    beta2.from=temp.substr(0, pos);
+    beta2.to=crypto; 
+    beta2.bid_price=info.bid_price;
+    beta2.bid_quantity=info.bid_quantity; 
+    return {beta1, beta2};
+}
+
+void ConvertToPriceInfoBeta::search_crypto_valut(std::vector<PriceInfo>& price_infos, std::string& crypto, std::vector<std::string>& crypto_name, std::unordered_set<std::string>& crypto_set){
```

Comment:
`crypto` parameter is not needed - its always equal to BTC. remove parameter from function declaration and use "BTC" string directly in line
```
         auto pos = info.symbol.find("BTC");  
```

Reply:
solved